### DISCUSSION TITLE
AWS Lambda & sqlite

### DISCUSSION THREAD

**Discussion by julesbouton** [sqlpage/SQLPage#537] [Q&A] [answered]
Hi,
Thanks (again!) for all the improvements on SQLpage. I am now trying to deploy my small project and I think I figured out how to do some stuff with a lambda function. Unfortunately, the documentation is a bit elliptic on this point.

Once the lambda zip file is ready, the documentation only mentions that we can put our .sql files in the zip folder (which so far works well). My question(s) are:

- Can I also put a sqlite file in the zip folder and use it as a database?
- Can the lambda file also contain custom components?

Thanks in advance for your help.
Best

**Comment by julesbouton** [comment]
So if anyone else is wondering, this is perfectly possible. The AWS lambda zip file contains exactly the same kind of files as when running in local.

**Reply by lovasoa** [reply]
Yes, it’s possible to include a SQLite file in your Lambda zip folder, but keep in mind that Lambda’s storage is ephemeral. This means that the data will be reset every time the function stops or scales, which you can't control.

**Comment by lovasoa** [comment, marked as answer]
Hi! 

Glad you like the latest changes! There is more to come ;)

While you can technically include a SQLite file in your Lambda zip folder, it's not the best fit for Lambda's environment. Lambda functions operate with ephemeral storage, meaning any data written to disk during execution is lost when the function ends. This makes it tricky to maintain a persistent SQLite database since the data would only persist during a single invocation of the function.

Instead, it’s generally better to use a managed database service, with a database system that supports remote connections, like Amazon RDS, or Aurora which offers persistent storage and better scalability. These services are designed to work well with Lambda, ensuring your data is available across multiple invocations and functions. You can also use an external database provider like TiDB Cloud or Neon.

As for custom components, yes, you can include them in your Lambda zip file! Just ensure they’re in `sqlpage/templates`. Keep in mind that everything your function needs to run must be included in the zip file.

**Reply by julesbouton** [reply]
Thanks for the details! For the moment, I only want to display content (ie = no need for the changes to persist), so I can stay with SQLite (which is a bit easier to manage). But good to know if others have the same question!

**Comment by lovasoa** [comment]
If you don't want to switch database systems and want an easy solution to quickly deploy your SQLPage app with SQLite, I of course recommend our own solution: https://beta.datapage.app !
